src/main/java/codeu/model/store/persistence/PersistentDataStore.java

```diff
@@ -200,6 +197,24 @@ public PersistentDataStore() {
     return activities;
   }
 
+  public List<Setting> loadSettings()throws PersistentDataStoreException{
+    List<Setting> setting = new ArrayList<>();
+    Query query = new Query("settings");
+    PreparedQuery results = datastore.prepare(query);
+
+    for(Entity entity: results.asIterable()){
+      try{
+      Setting setting1 = new Setting(Setting.SettingType.getSettingType((String) entity.getProperty("type")),(String)entity.getProperty("value"));
```

Comment:
Lets pull these parameters out into variables. Also we should null check against the the type since getSettingType can return null.

Reply:
got it.